lib/core/src/sdk.rs

```diff
@@ -1431,6 +1440,118 @@ impl LiquidSdk {
             }
         }
 
+        // TODO Ensure this happens only on first sync
```

Comment:
Perhaps it is better for now to put all logic of recovering chain data in its own function (recover_swaps_chain_data ?). I am thinking that because it might need to be reusable code and we still not sure how we should incorporate it.